### PR TITLE
🗃️ Fix Dates Off By One Pt 1 of 2 (#643) 

### PR DESCRIPTION
**Pull request by mistryrn**
## Description of Changes

PR 1 of 2 for fixing the `dateOfBirth` off by one issue by storing dates as `YYYY-MM-DD` strings. Adds a temporary `dateOfBirthString` column to `pi-das`, adds a migration and a data migration to populate that column, and a `package.json` script in `pi-das` to run that data migration.

### PI DAS
Add a temporary `dateOfBirthString` column to the PI DAS, containing `YYYY-MM-DD` versions of date of birth values.
- Updates `pi-das` schema to include temp `dateOfBirthString` column
- Adds migration to create new column
- Adds data migration script [pi-das/prisma/migrations/20240719155709_date_of_birth_string_pt_1/data-migration.ts](https://github.com/OHCRN/platform/commit/914183ef640212867dbdf1ecc9197779d464861d#diff-e409c300b2699c4d400992a6f94ecbb305c80d4efac6fb189513412722504807) to populate the new column with `YYYY-MM-DD` string-formatted versions of the existing `dateOfBirth` values
- Adds `pnpm prisma:migrate:data` script to PI DAS `package.json` scripts for running the new data migration

### Special Instructions
Before running these changes, you will need to first run the Prisma migration in PI DAS:
```bash
# from PI DAS
pnpm prisma:migrate:dev
```
Then, run the data migration in PI DAS:
```bash
# from PI DAS
pnpm prisma:migrate:data
```

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

File: apps/pi-das/tsconfig.json [hunk 1]
```diff
@@ -10,6 +10,6 @@
 		"esm": true,
 		"logError": true
 	},
-	"include": ["src", "prisma/seed"],
+	"include": ["src", "prisma/seed", "prisma/migrations/**/*.ts"],
```
Comment (mistryrn) [inline apps/pi-das/tsconfig.json:13]: Enables the tsconfig for data migration scripts 🙏

File: apps/pi-das/package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
 		"dev": "nodemon",
 		"lint": "eslint .",
 		"prisma:generate": "pnpm prisma generate",
+		"prisma:migrate:data": "ts-node ./prisma/migrations/20240719155709_date_of_birth_string_pt_1/data-migration.ts",
```
Comment (mistryrn) [inline apps/pi-das/package.json:15]: This is temporary, to make it so we can easily trigger the data migration within the PI DAS pod once this PR is deployed.

In PR 2 of 2 for this fix, this script will be removed from the `package.json` scripts.

File: apps/pi-das/prisma/schema.prisma [hunk 3]
```diff
@@ -31,6 +31,7 @@ model Participant {
   inviteId                    String?
   clinicianInvite             ClinicianInvite? @relation(fields: [inviteId], references: [id], onDelete: Restrict, onUpdate: Restrict)
   dateOfBirth                 DateTime         @db.Date
+  dateOfBirthString           String?
```
Comment (anncatton) [inline apps/pi-das/prisma/schema.prisma:34]: i can't remember what the order of operations will be in the part 2 pr, but would it be useful to put a char limit on the final version of the date of birth column?

**Review by anncatton — APPROVED** [review]
ran this part locally, is working 🌈

**Pull request merged**